File: app/src/main/java/kr/saintdev/pdiary/libs/data/DiaryObject.java
package kr.saintdev.pdiary.libs.data;

import java.util.Calendar;

public class DiaryObject {
    private int uniqid = 0;
    private String question = null;
    private String content = null;
    private Calendar date = null;

    public DiaryObject(int uniqid, String question, String content, Calendar date) {
        this.uniqid = uniqid;
        this.question = question;
        this.content = content;
        this.date = date;
    }

    public int getUniqid() {
        return uniqid;
    }

    public String getQuestion() {
        return question;
    }

    public String getContent() {
        return content;
    }

    public Calendar getDate() {
        return date;
    }
}

File: app/src/main/java/kr/saintdev/pdiary/libs/data/DiaryObject.kt
package kr.saintdev.pdiary.libs.data

import java.text.SimpleDateFormat
import java.time.LocalDate
import java.util.*

/**
 * 11.14 2018 (c) 2015-2019 Saint software All rights reserved.
 * 이 객체는 일기 한 편을 구현합니다.
 */
data class DiaryObject(
    val uniqid: Int,
    val question: String,
    val content: String?,
    val date: Calendar
)

fun Calendar.makeMMDD() : String {
    return "${this.get(Calendar.MONTH) + 1}/${this.get(Calendar.DAY_OF_MONTH)}"
}

fun String.makeSQLDate() : Calendar {
    val cal = Calendar.getInstance()
    val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.KOREAN)
    cal.time = sdf.parse(this)
    return cal
}